src/components/nowPlayingBar/nowPlayingBar.js

```diff
@@ -524,54 +524,46 @@ import { appRouter } from '../appRouter';
             height: imgHeight
         })) : null;
 
-        let isRefreshing = false;
-
-        if (url !== currentImgUrl) {
-            currentImgUrl = url;
-            isRefreshing = true;
-
-            if (url) {
-                imageLoader.lazyImage(nowPlayingImageElement, url);
-                nowPlayingImageElement.style.display = null;
-                nowPlayingTextElement.style.marginLeft = null;
-            } else {
-                nowPlayingImageElement.style.backgroundImage = '';
-                nowPlayingImageElement.style.display = 'none';
-                nowPlayingTextElement.style.marginLeft = '1em';
-            }
+        if (url && url !== nowPlayingImageUrl) {
```

Comment:
With `url !== nowPlayingImageUrl`, poster is not updated when switching (in the queue) from a song without an image to a song with an image.
I think you can leave the previous logic without `isRefreshing`.

Reply:
>poster is not updated when switching (in the queue) from a song without an image to a song with an image.

Unable to reproduce from a clean dev deployment (assuming we are referencing the image in the bottom left, within the nowPlayingBar). `url` is [assigned further up](src/components/nowPlayingBar/nowPlayingBar.js), so I would not expect such behavior?

Logic in this PR should be functionally the same as before. `nowPlayingImageUrl` is simply a rename of `currentImgUrl`. I took the opportunity to flatten
```
if (url !== currentImgUrl) {
  currentImgUrl = url;
  isRefreshing = true;
  if (url) {
```

into
```
if (url && url !== nowPlayingImageUrl) {
  nowPlayingImageUrl = url;
```
for readability - the nested scope is no longer necessary.

Would appreciate instructions / images to assist with reproducing this issues if we still think it is relevant.

Reply:
Logic is not the same. You don't update `nowPlayingImage` when `url` is `null`.

I'm not sure if I was just switching to the next one or switching back and forth. This will probably show an error:
1. Start song without an image.
2. Add song with image to the queue and move it right after the first one.
3. Ensure the nowPlayingBar is visible (open home).
4. Click `Next item` button.
Song 2 starts, image updates.
6. Click `Previous item` button.
Song 1 starts, image removed.
7. Click `Next item` button again.
Song 2 starts, but no image because `url` is equal to the cached one.

Reply:
Looks like the issue was specifically related to navigating back to the same song with an image.
Latest commit should fix the aforementioned issue - `nowPlayingImageUrl ` gets updated to `null` when playing a song with no image.

Could do the assignment after the conditional - check on `url`, but I prefer to do it inside to make intent clear.

Reply:
Currently it will execute the "null" block if the url has not been changed. This probably does no harm right now, simply because the update function is not called twice for the same item.

So I would leave the whole block as it was (just remove `isRefreshing` part) to avoid possible bug (with sequential call) and to minimize the count of changes. 🤷‍♂️

Reply:
Expanded conditional again so the DOM update only happens if `url !== nowPlayingImageUrl`.

Would rather avoid introducing `isRefreshing` again as it seems redundant both for readability and functionality.
I've aslso updated PR #3877